mts_bloggingbox/options/demo-importer/demo-files/news/theme_options.txt
###a:83:{s:8:"last_tab";s:1:"0";s:8:"mts_logo";s:0:"";s:11:"mts_favicon";s:0:"";s:14:"mts_touch_icon";s:0:"";s:14:"mts_metro_icon";s:0:"";s:20:"mts_twitter_username";s:0:"";s:14:"mts_feedburner";s:0:"";s:15:"mts_header_code";s:0:"";s:18:"mts_analytics_code";s:0:"";s:23:"mts_pagenavigation_type";s:1:"2";s:15:"mts_ajax_search";s:1:"0";s:14:"mts_responsive";s:1:"1";s:7:"mts_rtl";s:1:"0";s:17:"mts_shop_products";s:1:"9";s:15:"mts_prefetching";s:1:"0";s:13:"mts_lazy_load";s:1:"0";s:20:"mts_lazy_load_thumbs";s:1:"0";s:21:"mts_lazy_load_content";s:1:"0";s:12:"mts_async_js";s:1:"1";s:21:"mts_remove_ver_params";s:1:"1";s:15:"mts_optimize_wc";s:1:"1";s:16:"mts_color_scheme";s:7:"#cd3131";s:17:"mts_color_scheme2";s:7:"#141829";s:10:"mts_layout";s:8:"sclayout";s:14:"mts_background";a:10:{s:5:"color";s:7:"#eeeeee";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:0:"";s:10:"attachment";s:0:"";s:8:"position";s:0:"";s:4:"size";s:0:"";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}s:8:"parallax";s:1:"0";}s:14:"mts_custom_css";s:0:"";s:12:"mts_lightbox";s:1:"0";s:14:"mts_sticky_nav";s:1:"1";s:20:"mts_show_primary_nav";s:1:"1";s:15:"mts_header_cart";s:1:"0";s:17:"mts_header_search";s:1:"1";s:19:"mts_header_section2";s:1:"1";s:13:"mts_header_bg";a:10:{s:5:"color";s:7:"#cd3131";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:6:"repeat";s:10:"attachment";s:6:"scroll";s:8:"position";s:8:"left top";s:4:"size";s:5:"cover";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}s:8:"parallax";s:1:"0";}s:13:"mts_footer_bg";a:10:{s:5:"color";s:7:"#252525";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:6:"repeat";s:10:"attachment";s:6:"scroll";s:8:"position";s:8:"left top";s:4:"size";s:5:"cover";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}s:8:"parallax";s:1:"0";}s:22:"mts_footer_logo_button";s:1:"1";s:15:"mts_footer_logo";s:0:"";s:20:"mts_social_icon_head";s:1:"1";s:17:"mts_footer_social";a:4:{i:0;a:7:{s:11:"group_title";s:8:"Facebook";s:10:"group_sort";s:1:"1";s:21:"mts_footer_icon_title";s:8:"Facebook";s:15:"mts_footer_icon";s:8:"facebook";s:20:"mts_footer_icon_link";s:1:"#";s:21:"mts_footer_icon_color";s:7:"#aaaaaa";s:27:"mts_footer_icon_hover_color";s:7:"#aaaaaa";}i:1;a:7:{s:11:"group_title";s:7:"Twitter";s:10:"group_sort";s:1:"2";s:21:"mts_footer_icon_title";s:7:"Twitter";s:15:"mts_footer_icon";s:7:"twitter";s:20:"mts_footer_icon_link";s:1:"#";s:21:"mts_footer_icon_color";s:7:"#aaaaaa";s:27:"mts_footer_icon_hover_color";s:7:"#aaaaaa";}i:2;a:7:{s:11:"group_title";s:11:"Google Plus";s:10:"group_sort";s:1:"3";s:21:"mts_footer_icon_title";s:11:"Google Plus";s:15:"mts_footer_icon";s:11:"google-plus";s:20:"mts_footer_icon_link";s:1:"#";s:21:"mts_footer_icon_color";s:7:"#aaaaaa";s:27:"mts_footer_icon_hover_color";s:7:"#aaaaaa";}i:3;a:7:{s:11:"group_title";s:9:"Instagram";s:10:"group_sort";s:1:"4";s:21:"mts_footer_icon_title";s:9:"Instagram";s:15:"mts_footer_icon";s:9:"instagram";s:20:"mts_footer_icon_link";s:1:"#";s:21:"mts_footer_icon_color";s:7:"#aaaaaa";s:27:"mts_footer_icon_hover_color";s:7:"#aaaaaa";}}s:11:"mts_readers";s:1:"1";s:17:"mts_readers_count";s:8:"37676651";s:16:"mts_readers_text";s:16:"visitors / month";s:16:"mts_first_footer";s:1:"0";s:20:"mts_first_footer_num";s:1:"5";s:14:"mts_copyrights";s:73:"Theme by <a href="http://mythemeshop.com/" rel="nofollow">MyThemeShop</a>";s:17:"mts_header_layout";a:1:{s:7:"enabled";a:2:{s:6:"slider";s:6:"Slider";s:3:"cat";s:19:"Featured Categories";}}s:19:"mts_featured_slider";s:1:"0";s:23:"mts_featured_slider_num";s:1:"4";s:29:"mts_featured_category_section";s:1:"1";s:20:"mts_featured_section";a:5:{i:0;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"1";s:17:"mts_featured_cats";s:1:"2";s:23:"mts_featured_text_color";s:7:"#ffffff";s:24:"mts_featured_hover_color";s:7:"#cd3131";}i:1;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"2";s:17:"mts_featured_cats";s:1:"3";s:23:"mts_featured_text_color";s:7:"#ffffff";s:24:"mts_featured_hover_color";s:7:"#cd3131";}i:2;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"3";s:17:"mts_featured_cats";s:1:"4";s:23:"mts_featured_text_color";s:7:"#ffffff";s:24:"mts_featured_hover_color";s:7:"#cd3131";}i:3;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"4";s:17:"mts_featured_cats";s:1:"5";s:23:"mts_featured_text_color";s:7:"#ffffff";s:24:"mts_featured_hover_color";s:7:"#cd3131";}i:4;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"5";s:17:"mts_featured_cats";s:1:"6";s:23:"mts_featured_text_color";s:7:"#ffffff";s:24:"mts_featured_hover_color";s:7:"#cd3131";}}s:15:"mts_featured_bg";a:10:{s:5:"color";s:7:"#141829";s:3:"use";s:7:"pattern";s:13:"image_pattern";s:4:"nobg";s:12:"image_upload";s:0:"";s:6:"repeat";s:6:"repeat";s:10:"attachment";s:6:"scroll";s:8:"position";s:8:"left top";s:4:"size";s:5:"cover";s:8:"gradient";a:3:{s:4:"from";s:7:"#ffffff";s:2:"to";s:7:"#000000";s:9:"direction";s:10:"horizontal";}s:8:"parallax";s:1:"0";}s:23:"mts_featured_categories";a:1:{i:0;a:5:{s:11:"group_title";s:0:"";s:10:"group_sort";s:1:"1";s:21:"mts_featured_category";s:6:"latest";s:28:"mts_featured_category_layout";s:8:"gridpost";s:30:"mts_featured_category_postsnum";s:1:"6";}}s:15:"mts_enable_tabs";s:1:"0";s:27:"mts_popular_tab_posts_title";s:12:"Most Popular";s:25:"mts_popular_tab_posts_num";s:1:"6";s:27:"mts_popular_tab_posts_limit";s:3:"300";s:25:"mts_home_meta_info_enable";a:5:{s:4:"time";s:1:"1";s:12:"author-image";s:1:"1";s:6:"author";s:1:"1";s:7:"comment";s:1:"1";s:8:"readmore";s:1:"1";}s:22:"mts_single_post_layout";a:1:{s:7:"enabled";a:4:{s:7:"content";s:12:"Post Content";s:6:"author";s:10:"Author Box";s:12:"social-share";s:19:"Social Share & Tags";s:13:"previous-next";s:21:"Previous & Next Posts";}}s:8:"mts_tags";s:1:"0";s:27:"mts_single_meta_info_enable";a:5:{s:8:"category";s:1:"1";s:12:"author-image";s:1:"1";s:6:"author";s:1:"1";s:4:"time";s:1:"1";s:7:"comment";s:1:"1";}s:14:"mts_breadcrumb";s:1:"1";s:16:"mts_like_dislike";s:1:"1";s:18:"mts_author_comment";s:1:"1";s:16:"mts_comment_date";s:1:"1";s:18:"mts_social_buttons";a:2:{s:7:"enabled";a:5:{s:13:"facebookshare";s:14:"Facebook Share";s:8:"facebook";s:13:"Facebook Like";s:7:"twitter";s:7:"Twitter";s:5:"gplus";s:11:"Google Plus";s:9:"pinterest";s:9:"Pinterest";}s:8:"disabled";a:2:{s:8:"linkedin";s:8:"LinkedIn";s:7:"stumble";s:11:"StumbleUpon";}}s:18:"mts_posttop_adcode";s:0:"";s:23:"mts_posttop_adcode_time";s:1:"0";s:18:"mts_postend_adcode";s:0:"";s:23:"mts_postend_adcode_time";s:1:"0";s:20:"mts_sidebar_for_home";s:0:"";s:20:"mts_sidebar_for_post";s:0:"";s:20:"mts_sidebar_for_page";s:0:"";s:23:"mts_sidebar_for_archive";s:0:"";s:24:"mts_sidebar_for_category";s:0:"";s:19:"mts_sidebar_for_tag";s:0:"";s:20:"mts_sidebar_for_date";s:0:"";s:22:"mts_sidebar_for_author";s:0:"";s:22:"mts_sidebar_for_search";s:0:"";s:24:"mts_sidebar_for_notfound";s:0:"";s:20:"mts_sidebar_for_shop";s:0:"";s:23:"mts_sidebar_for_product";s:0:"";s:19:"mts_typography_sets";a:1:{i:0;s:5:"latin";}s:29:"google_typography_collections";a:20:{i:0;a:11:{s:12:"preview_text";s:9:"Logo Font";s:13:"preview_color";s:4:"dark";s:11:"font_family";s:9:"Quicksand";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"30px";s:10:"font_color";s:7:"#ffffff";s:13:"css_selectors";s:22:"#header h1, #header h2";s:14:"additional_css";s:0:"";s:11:"backup_font";s:0:"";s:7:"default";s:4:"true";s:16:"collection_title";s:9:"Logo Font";}i:1;a:11:{s:12:"preview_text";s:15:"Navigation Font";s:13:"preview_color";s:4:"dark";s:11:"font_family";s:9:"Quicksand";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"16px";s:10:"font_color";s:7:"#ffffff";s:13:"css_selectors";s:55:"#primary-navigation a, #header .mts-cart .cart-contents";s:14:"additional_css";s:0:"";s:11:"backup_font";s:0:"";s:7:"default";s:4:"true";s:16:"collection_title";s:15:"Navigation Font";}i:2;a:11:{s:12:"preview_text";s:18:"Home Article Title";s:13:"preview_color";s:5:"light";s:11:"font_family";s:9:"Quicksand";s:12:"font_variant";s:3:"500";s:9:"font_size";s:4:"26px";s:10:"font_color";s:7:"#141829";s:13:"css_selectors";s:119:".l

mts_bloggingbox/changelog.txt
v1.2.8 (Dec 23, 2020)
-----------------------------
- Fixed compatibility issues with the WordPress 5.6
- Fixed comment reply functionality
- Fixed page jump when user clicks on the description tab in the single product
- Added 8 new Google fonts in the typography options

Files Updated:
- comments.php
- functions.php
- functions/rank-math-notice.php
- functions/rm-seo.php
- js/customscript.js
- lang/bloggingbox.pot
- options/css/aristo.css
- options/fields/button_set/field_button_set.php
- options/fields/button_set_hide_below/field_button_set_hide_below.php
- options/fields/multi_checkbox/field_multi_checkbox.php
- options/fields/background/field_background.php
- options/js/options.js
- style.css


v1.2.7 (Aug 19, 2020)
-----------------------------
- Fixed compatibility issues with the WordPress 5.5
- Disabled WP 5.5 core lazy loading if theme's lazy loading option is enabled
- Added 9 new Google fonts in the typography options

Files Updated:
- functions.php
- options/fields/group/field_group.js
- options/fields/layout2/field_layout2.js
- options/fields/multi_text/field_multi_text.js
- lang/bloggingbox.pot
- theme-options.php
- style.css


v1.2.6 (Apr 15, 2020)
----------------------------
- Options Panel will work even without the MyThemeShop Updater plugin, but you will not receive any future updates without this plugin
- Added 10 new Google fonts in the typography options

Files Updated:
- functions.php
- functions/plugin-activation.php
- functions/theme-actions.php
- lang/bloggingbox.pot
- style.css


v1.2.5 (April 7, 2020)
-----------------------------
- Released BloggingBox premium theme for free
- Fixed 'data-vocabulary' schema warning in the breadcrumbs
- Added 7 new Google fonts in the typography options

Files Updated:
- functions.php
- functions/plugin-activation.php
- functions/rm-seo.php
- functions/rank-math-notice.php
- functions/theme-actions.php
- lang/bloggingbox.pot
- page-templates/contact.php
- page.php
- single.php
- style.css


v1.2.4 (Nov 15, 2019)
-----------------------------
- Fixed notices related to WP 5.3

Files Updated:
- functions.php
- lang/bloggingbox.pot
- options/options.php
- style.css


v1.2.3 (Jul 17, 2019)
-----------------------------
- Fixed 'Upload Theme' button not working
- Added 15 new Google fonts in the typography options

 Files Updated:
- functions/plugin-activation.php
- functions/rm-seo.php
- functions.php
- options/css/options.css
- style.css


v1.2.2 (Jan 12, 2019)
-----------------------------
- Updated the Rank Math's auto installation process to give the control back to the users
- Fixed MyThemeShop Connect install issue on Recommended Plugins page

Files Updated:
- functions.php
- functions/rank-math-notice.php
- functions/rm-seo.php
- functions/theme-actions.php
- images/rank-math-banner.jpg
- lang/bloggingbox.pot
- style.css


v1.2.1 (Nov 22, 2018)
-----------------------------
- Added More SEO Tweaks
- Added better integration with Rank Math SEO plugin
- Added integration for Rank Math SEO Breadcrumbs
- Fixed Twitter widget issue
- Replaced inline transparent 1x1 image

Files Updated:
- comments.php
- functions.php
- single.php
- style.css
- functions/cacert.pem
- functions/rm-seo.php
- functions/rm-seo.php
- functions/twitteroauth.php
- functions/widget-tweets.php
- lang/bloggingbox.pot
- page-templates/contact.php
- page.php
- single.php


v1.2.0 (Oct 8, 2018)
-----------------------------
- Added life-changing SEO support with an included SEO plugin
- Added compatibility for Gutenberg editor
- Added 6 new Google fonts in the typography options
- Security & licensing fixes
- Fixed WooCommerce Product Search button styling issue
- Fixed WooCommerce Layered Nav Dropdown button styling issue
- Fixed other minor CSS issues
- Important fixes for better performance

Files Updated:
- functions.php
- style.css
- css/woocommerce2.css
- functions/plugin-activation.php
- functions/rm-seo.php
- functions/theme-actions.php
- options/options.php
- lang/bloggingbox.pot


v1.1.0 (Sep 06, 2018)
-----------------------------
- Fixed Google structured data issues for blog posts

Files Updated:
- style.css
- functions.php

mts_bloggingbox/options/js/options.js
var changeWarn = false;
jQuery(document).ready(function($){
	var History = window.History;
	function loadTabs(){
		jQuery('.nhp-opts-group-tab').hide();
		jQuery('.nhp-opts-group-tab-link-li').removeClass('active');
		if(jQuery('#last_tab').val() == ''){
			jQuery('.nhp-opts-group-tab:first').slideDown('fast');
			jQuery('#nhp-opts-group-menu li:first').addClass('active');
		}else{
			tabid = jQuery('#last_tab').val();
			jQuery('#'+tabid+'_section_group').slideDown('fast');
			jQuery('#'+tabid+'_section_group_li').addClass('active');
			if (jQuery('#'+tabid+'_section_group_li').closest('#nhp-opts-homepage-accordion').length)
				jQuery('#nhp-opts-homepage-accordion').show();
			if (jQuery('#'+tabid+'_section_group_li').closest('#nhp-opts-blog-accordion').length)
				jQuery('#nhp-opts-blog-accordion').show();
		}
	}
	jQuery('.nhp-opts-group-tab-link-a').click(function(){
		clear_search();

		if (this.id == 'accordion_section_group_li_a') {
		   jQuery('#nhp-opts-homepage-accordion').slideToggle();
		   return false;
		}
		if (this.id == 'accordion_section_group_li_a_2') {
		   jQuery('#nhp-opts-blog-accordion').slideToggle();
		   return false;
		}
		var $this = jQuery(this);
		if ($this.parent().hasClass('active'))
			return false;
		var relid = jQuery(this).attr('data-rel');
		
		jQuery('.nhp-opts-group-tab').hide();
		jQuery('#'+relid+'_section_group').fadeIn(400);

		jQuery('.nhp-opts-group-tab-link-li').removeClass('active');
		$this.parent().addClass('active');

		//jQuery("html, body").animate({ scrollTop: jQuery('#nhp-opts-header').offset().top - 48 }, 500);

		History.pushState( {tab:relid}, document.title, "themes.php?page=theme_options&tab="+relid );

		return false;
	});

	History.Adapter.bind( window,'load statechange', function( event ){
		var State = History.getState();
		var tab = State.data.tab;

		if ( typeof tab !== 'undefined' ) {
			jQuery('#'+tab+'_section_group_li_a').click();
		} else {
			loadTabs();
		}
	});
	jQuery('#nhp-opts-form-wrapper').submit(function(){
		var currentTabID = jQuery(this).find('#nhp-opts-group-menu li.active a').attr('data-rel');
		jQuery('#last_tab').val(currentTabID);
	});
	

	if(jQuery('#nhp-opts-save').is(':visible')){
		jQuery('#nhp-opts-save').delay(4000).slideUp('slow');
	}
	function addTypographyData(container, form) {
		var i = 0;
		container.find(".collections .collection").each(function() {
			form.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][preview_text]', jQuery(this).find(".preview_text").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][preview_color]', jQuery(this).find(".preview_color li.current a").attr("class"))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_family]', jQuery(this).find(".font_family").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_variant]', jQuery(this).find(".font_variant").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_size]', jQuery(this).find(".font_size").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][font_color]', jQuery(this).find(".font_color").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][css_selectors]', jQuery(this).find(".css_selectors").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][additional_css]', jQuery(this).find(".additional_css").val())
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][backup_font]', jQuery(this).find(".backup_font").select2('val'))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][default]', jQuery(this).attr("data-default"))
				.addHiddenField(nhpopts.opt_name+'[google_typography_collections]['+i+'][collection_title]', jQuery(this).find(".collection_title").val());
			
			i++;
		});
	}
	
	if(jQuery('#nhp-opts-imported').is(':visible')){
		jQuery('#nhp-opts-imported').delay(4000).slideUp('slow');
	}
	
	jQuery('#nhp-opts-footer').find('#savechanges').click(function(e) {
		// add typography data via hidden fields before submitting
		if (typography_isloaded) {
			addTypographyData(jQuery('#google_typography'),jQuery('#nhp-opts-form-wrapper'));
		}
		// AJAX save
		/*
		jQuery('#savechanges').prop('disabled', true).after('<div class="spinner" id="ajax-saving"></div>');
		changeWarn = false;
		var $form = jQuery('#nhp-opts-form-wrapper');
		jQuery.post( $form.attr('action'), $form.serialize() ).done(function() {
			jQuery('#ajax-saving').remove();
			jQuery('#savechanges').prop('disabled', false).after('<div id="ajax-saved">Settings saved!</div>');
			setTimeout(function() { jQuery('#ajax-saved').fadeOut('slow', function() { jQuery('#ajax-saved').remove(); }); }, 2000);
		});
		return false;
		*/
		// AJAX save end
	});

	jQuery('input, textarea, select').change(function(){
		if (!changeWarn) {
		  //jQuery('#nhp-opts-save-warn').slideDown('slow');
		  changeWarn = true;
		}
	});

	jQuery('#nhp-opts-form-wrapper').submit(function() {
		changeWarn = false;
	});

	window.onbeforeunload = confirmExit;
	function confirmExit() {
		if (changeWarn) {
			return nhpopts.leave_page_confirm;
		}
	}
	
	jQuery('#nhp-opts-import-code-button').click(function(e){
		e.preventDefault();
		jQuery('#nhp-opts-import-code-wrapper').toggle().find('#import-code-value').val('');
	});

	jQuery('#nhp-opts-export-code-copy').click(function(e){
		e.preventDefault();
		jQuery('#nhp-opts-export-code').toggle().select();
	});
	
	// Presets
	function scrollImportLogToBottom(){
		var element = document.getElementById("importing-modal-content");
		element.scrollTop = element.scrollHeight;
	}
	jQuery('#presets .preset .import-demo-button, #presets .preset .import-demo-widgets-button, #presets .preset .import-demo-options-button').on('click', function(e) {
		e.preventDefault();

		var $this = jQuery(this);
		var $parent = $this.closest('.preset');
		var confirmText = nhpopts.import_opt_confirm;
		if ( $this.hasClass('import-demo-button') ) {
			confirmText = nhpopts.import_all_confirm;
		}
		if ( $this.hasClass('import-demo-widgets-button') ) {
			confirmText = nhpopts.import_widget_confirm;
		}

		var result = confirm( confirmText );
		if ( result ) {

			var data = {};
			data.action = "mts_install_demo";
			data.demo_import_id = $parent.attr("data-demo-id");
			data.nonce = $parent.attr("data-nonce");
			data.demo_import_options = '1';
			data.demo_import_content = '0';
			data.demo_import_widgets = '0';

			if ( $this.hasClass('import-demo-button') ) {
				data.demo_import_content = '1';
				data.demo_import_widgets = '1';
			}

			if ( $this.hasClass('import-demo-widgets-button') ) {
				data.demo_import_widgets = '1';
			}

			$this.magnificPopup({
				items: {
					src: '#importing-overlay',
					type: 'inline'
				},
				modal: true
			}).magnificPopup('open');

			var last_response_len = false;
			jQuery.ajax( ajaxurl, {
				data: data,
				xhrFields: {
					onprogress: function(e) {

						var this_response, response = e.currentTarget.response;
						if(last_response_len === false) {

							this_response = response;
							last_response_len = response.length;

						} else {

							this_response = response.substring(last_response_len);
							last_response_len = response.length;
						}

						jQuery('#importing-modal-content').append(this_response);
						scrollImportLogToBottom();
					}
				}
			})
			.done(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.import_done);
				jQuery('#importing-modal-footer-info').text(nhpopts.import_done);
				jQuery('#importing-modal-footer-button').show();
			})
			.fail(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.import_fail);
				jQuery('#importing-modal-footer-info').text(nhpopts.import_fail);
				jQuery('#importing-modal-footer-button').show();
			});
		}

		return false;
	});

	jQuery('.remove-demo-button').on('click', function(e) {
		e.preventDefault();

		var result = confirm( nhpopts.remove_all_confirm );
		if ( result ) {

			var $this = jQuery(this);
			var data = {};
			data.action = "mts_install_demo";
			data.mts_remove_demos = "1";
			data.nonce = $this.attr("data-nonce");

			$this.magnificPopup({
				items: {
					src: '#importing-overlay',
					type: 'inline'
				},
				modal: true
			}).magnificPopup('open');

			var last_response_len = false;
			jQuery.ajax( ajaxurl, {
				data: data,
				xhrFields: {
					onprogress: function(e) {

						var this_response, response = e.currentTarget.response;
						if(last_response_len === false) {

							this_response = response;
							last_response_len = response.length;

						} else {

							this_response = response.substring(last_response_len);
							last_response_len = response.length;
						}

						jQuery('#importing-modal-content').append(this_response);
						scrollImportLogToBottom();
					}
				}
			})
			.done(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.remove_done);
				jQuery('#importing-modal-footer-info').text(nhpopts.remove_done);
				jQuery('#importing-modal-footer-button').show();
			})
			.fail(function(data) {
				jQuery('#importing-modal-header h2').text(nhpopts.remove_fail);
				jQuery('#importing-modal-footer-info').text(nhpopts.remove_fail);
				jQuery('#importing-modal-footer-button').show();
			});
		}

		return false;
	});

	function mtsRemoveURLParameter(url, parameter) {

		//prefer to use l.search if you have a location/link object
		var urlparts= url.split('?');
		if ( urlparts.length >= 2 ) {

			var prefix= encodeURIComponent(parameter)+'=';
			var pars= urlparts[1].split(/[&;]/g);

			//reverse iteration as may be destructive
			for (var i= pars.length; i-- > 0;) {
				//idiom for string.startsWith
				if (pars[i].lastIndexOf(prefix, 0) !== -1) {
					pars.splice(i, 1);
				}
			}

			url= urlparts[0]+'?'+pars.join('&');
			return url;

		} else {

			return url;
		}
	}
	
	jQuery('#importing-modal-footer-button').on('click', function(e) {
		e.preventDefault();
		jQuery(this).prop('disabled', true ).text(nhpopts.reloading_page);
		var a = mtsRemoveURLParameter( window.location.href , 'tab' );
		window.location.href = a;
	});

	
	// Confirm import
	jQuery('#nhp-opts-import').click(function() {
		return confirm(nhpopts.import_confirm);
	});
	
	// Confirm reset
	jQuery('input[name="'+nhpopts.opt_name+'[defaults]"]').click(function() {
		return confirm(nhpopts.reset_confirm);
	});
	
	// Disallow submission by enter key
	jQuery('#nhp-opts-form-wrapper').find('input').keydown(function(event){
		if ( event.keyCode == 13 ){
			event.preventDefault();
		}
	});
	
	// Floating footer
	var $footer = jQuery('#nhp-opts-footer');
	var $bottom = jQuery('#nhp-opts-bottom');
	
	
	$footer.addClass('floating');
	jQuery(document).on('scroll', function(){
		if ($bottom.isOnScreen()) {
			$footer.removeClass('floating');
		} else {
			$footer.addClass('floating');
		}
	});
	if ($bottom.isOnScreen()) {
		$footer.removeClass('floating');
	}
	
	// Needs JS sizing when position:fixed
	var footer_padding = $footer.innerWidth() - $footer.width();
	function resizeFloatingElements() {
		var w = jQuery('#nhp-opts-form-wrapper').width();
		$footer.width(w - footer_padding);
	}
	resizeFloatingElements();
	
	var resizeTimer;
	jQuery(window).resize(function() {
		clearTimeout(resizeTimer);
		resizeTimer = setTimeout(resizeFloatingElements, 100);
	});

	// Child theme creator
	jQuery('#nhp-opts-child-button').on('click', function(e) {
		e.preventDefault();
		var elem		 = jQuery(this),
			childName	= jQuery('#nhp-opts-child-name'),
			childNameVal = childName.val();

		if ( '' === childNameVal ) {
			alert( nhpopts.child_theme_name_empty );
		} else {
			jQuery.ajax({
				url: ajaxurl, 
				method: 'post',
				data: {
					'action' : 'mts_child_theme',
					'child_name' : childNameVal
				},
				beforeSend: function() {
					elem.prop('disabled', true);
					childName.prop('disabled', true);
				},
				success: function(data) {
					elem.prop('disabled', false);
					childName.val('').prop('disabled', false);
					mtsUpdateChildThemesList();
				}
			});
		}
	});
	// Refresh the existing child themes list
	function mtsUpdateChildThemesList() {
		jQuery.post( ajaxurl, { action: 'mts_list_child_themes' }, function( response ) {
			if ( response ) {
				jQuery('#child-theme-list-wrap').html( response );
			}
		});
	}
	
	$('#search-theme-options').keydown(function(e) {
		if (e.keyCode == 27) {
			clear_search( true );
			$(this).val('').blur();
			return false;
		}
	}).keyup(function (e) {
		var value = $(this).val();
		if ( value == '' ) {
			clear_search( true );
			return false;
		}
		search_options( value );
	}).blur(function(event) {
		var value = $(this).val();
		if ( value == '' ) {
			$(this).closest('div').removeClass('active');
		}
	});
	$('.options-search-link').click(function(event) {
		event.preventDefault();
		if ($(this).closest('div').hasClass('active')) {
			var query = $(this).closest('div').addClass('active').find('input').val();
			search_options( query );
		}
		$(this).closest('div').addClass('active').find('input').focus();
	});
	$('.theme-options-clear-search').click(function(event) {
		event.preventDefault();
		clear_search( true );
		$(this).siblings('input').val('').trigger('blur');
	});
	$('#nhp-opts-header .docsupport').click(function(event){
		event.preventDefault();
		$('.nhp-opts-group-tab-link-a[data-rel=support]').trigger('click');
	});

	// Reverse jQuery plugin
	jQuery.fn.reverse = [].reverse;

	var searched = false;
	function search_options( query ) {
		searched = true;
		// de-select tabs
		$('.nhp-opts-group-tab-link-li.active').removeClass('active');
		$('.nhp-opts-info-field').hide();
		query = query.toLowerCase();
		var noresults = true;
		$('#nhp-opts-main').children('div').hide().each(function(index, el) {
			if (this.id == 'typography_default_section_group') {
				return true;
			}
			var $contents = $(this);

			//var $title = $contents.find('h2');
			//var $desc = $contents.find('.nhp-opts-section-desc');
			
			$contents.children('table').children('tbody').children('tr').each(function(index, el) {
				var $row = $(this);
				$row.hide();
				if ($row.find('th').text().toLowerCase().indexOf( query ) !== -1) {
					$contents.show();
					$row.show();
					noresults = false;
				}

			}).reverse().each(function(index, el) {
				if ($(this).is(':visible') && $(this).find('.buttonset-hide').length) {
					$(this).find('#nhp-opts-button-show-below').each(function(event){
						if (jQuery(this).hasClass('ui-state-active')) {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').show();
						} else {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').hide();
						}
					});
				}
			});
		});
		if (noresults) {
			$('#options-search-no-results').show();
		}
	}

	function clear_search( clicktab ) {
		if ( ! searched ) {
			return false;
		}
		$('#nhp-opts-main').children().each(function(index, el) {
			$(this).find('tr').show().reverse().each(function(index, el) {
				if ($(this).find('.buttonset-hide').length) {
					$(this).find('#nhp-opts-button-show-below').each(function(event){
						if (jQuery(this).hasClass('ui-state-active')) {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').show();
						} else {
							var num = jQuery(this).parent().data('hide');
							jQuery(this).closest('tr').nextAll('tr:lt('+num+')').hide();
						}
					});
				}
			});
		});
		$('.nhp-opts-info-field').show();
		if ( clicktab ) {
			$('#nhp-opts-main').children().hide().first().show();
			$('#0_section_group_li').addClass('active');
		} else {
			$('#search-theme-options').val('').parent().removeClass('active');
		}
		searched = false;
	}
});

jQuery(window).on('load', function() {
	jQuery('#savechanges').prop('disabled', false);
});

jQuery.fn.isOnScreen = function(){

	var win = jQuery(window);

	var viewport = {
		top : win.scrollTop(),
		left : win.scrollLeft()
	};
	viewport.right = viewport.left + win.width();
	viewport.bottom = viewport.top + win.height();

	var bounds = this.offset();
	bounds.right = bounds.left + this.outerWidth();
	bounds.bottom = bounds.top + this.outerHeight();

	return (!(viewport.right < bounds.left || viewport.left > bounds.right || viewport.bottom < bounds.top || viewport.top > bounds.bottom));

};

jQuery.fn.addHiddenField = function(name, value) {
	this.each(function () {
		var elem_id = name.replace(/\W/g, '-').replace(/--+/g, '-').replace(/(^-|-$)/, '');
		if (jQuery('#'+elem_id).length) {
			// elem exists, change value
			jQuery('#'+elem_id).val(value);
		} else {
			// elem doesn't exist, create
			var input = jQuery("<input>").attr("type", "hidden").attr("id", elem_id).attr("name", name).val(value);
			jQuery(this).append(jQuery(input));
		}
		
	});
	return this;
};

var fnDelay = (function(){
  var timer = 0;
  return function(callback, ms){
	clearTimeout (timer);
	timer = setTimeout(callback, ms);
  };
})();
